core/src/main/scala-3/caliban/schema/AnnotationsVersionSpecific.scala

```diff
@@ -12,4 +14,14 @@ trait AnnotationsVersionSpecific {
    */
   case class GQLField() extends StaticAnnotation
 
+  /**
+   * Annotation used to provide directives to a schema type
+   */
+  open class GQLDirective(val directive: Directive) extends StaticAnnotation
```

Comment:
Made this an `open class` in Scala 3, so that users can extend it without having to add the `import scala.language.adhocExtensions` import. More info on open classes [here](https://docs.scala-lang.org/scala3/reference/other-new-features/open-classes.html)